library/proc_macro/src/bridge/handle.rs

```diff
@@ -51,31 +53,15 @@ impl<T> IndexMut<Handle> for OwnedStore<T> {
     }
 }
 
-// HACK(eddyb) deterministic `std::collections::hash_map::RandomState` replacement
-// that doesn't require adding any dependencies to `proc_macro` (like `rustc-hash`).
-#[derive(Clone)]
-struct NonRandomState;
-
-impl BuildHasher for NonRandomState {
-    type Hasher = std::collections::hash_map::DefaultHasher;
-    #[inline]
-    fn build_hasher(&self) -> Self::Hasher {
-        Self::Hasher::new()
-    }
-}
-
 /// Like `OwnedStore`, but avoids storing any value more than once.
 pub(super) struct InternedStore<T: 'static> {
     owned: OwnedStore<T>,
-    interner: HashMap<T, Handle, NonRandomState>,
+    interner: FxHashMap<T, Handle>,
```

Comment:
Two thoughts here:
* any chance this is significant enough on its own to get tested in a separate PR?
* just realized that the "base index" trick could work here to allow a simple `Vec` backing storage - I guess the handle counters would need to become thread-local in that case?

Reply:
I'm not sure how valuable it would be in a separate PR given that the types being hashed are presumably simpler without this change, but I suppose we could test it if we wanted to. I've kept that part in a separate commit so it can easily be split off.

I think the simple `Vec` storage might only work for `InternedStore`, and it'd require a thread-local counter and only one server per-thread as well, but that's probably reasonable. It seems unlikely we'd get into a situation where a single client thread is shared by multiple different server threads, which could cause a setup like that to break for server-side interning.

The `Vec` approach with a non-atomic counter feels the most reliable for client-side unique name generation.